lib/active_fedora/file.rb

```diff
@@ -101,6 +107,7 @@ def datastream_will_change!
     def attribute_will_change!(attr)
       if attr == 'content'
         changed_attributes['content'] = true
+      elsif attr == 'type'
```

Comment:
@mjgiarlo maybe a guard clause for this nil case? or just explicitly return nil... dealer's choice 😉